ods_ci/tests/Resources/Page/ModelRegistry/ModelRegistry.resource

```diff
@@ -0,0 +1,400 @@
+    ${file_count}=    Get Length    ${cert_files}
+    Should Be Equal As Numbers    ${file_count}    13    The number of certificate files created should be 13
+    FOR    ${file}    IN    @{cert_files}
+        File Should Exist    ${file}
+    END
+
+Upload Certificate To Jupyter Notebook
+    [Documentation]    Uploads file to Jupyter Notebook
+    [Arguments]    ${domain_cert}
+    Upload File In The Workbench     filepath=${domain_cert}    workbench_title=${WORKBENCH_TITLE}
+    ...    workbench_namespace=${PRJ_TITLE}
+
+Create Generic Secret
+    [Documentation]    Creates Secret for model registry in a given namespace
+    [Arguments]    ${namespace}    ${secret_name}    ${key_file}    ${crt_file}    ${ca_file}
+    Log    This is the secret name ${secret_name}
+    ${command}=    Set Variable
+    ...    oc create secret -n ${namespace} generic ${secret_name} --from-file=tls.key=${key_file} --from-file=tls.crt=${crt_file} --from-file=ca.crt=${ca_file}    # robocop: disable:line-too-long
+    Run Process    ${command}    shell=True
+    Log    Secret ${secret_name}, namespace ${namespace}
+    ${output}=    Run Process    oc get secret ${secret_name} -n ${namespace}    shell=True
+    Should Contain    ${output.stdout}    ${secret_name}
+
+Apply Db Config Samples
+    [Documentation]    Applying the db config samples from https://github.com/opendatahub-io/model-registry-operator
+    [Arguments]    ${namespace}
+    ${rc}    ${out}=    Run And Return Rc And Output
+    ...    oc apply -k ${MODEL_REGISTRY_DB_SAMPLES} -n ${namespace}
+    Should Be Equal As Integers	  ${rc}	 0   msg=${out}
+
+Jupyter Notebook Can Query Model Registry
+    [Documentation]    Runs the test workbench and check if there was no error during execution
+    [Arguments]    ${filepath}
+    Open Notebook File In JupyterLab    ${filepath}
+    Open With JupyterLab Menu  Run  Restart Kernel and Run All Cells…
+    Wait Until Page Contains Element    xpath=//div[contains(text(),"Restart") and @class="jp-Dialog-buttonLabel"]
+    Click Element    xpath=//div[contains(text(),"Restart") and @class="jp-Dialog-buttonLabel"]
+    # Somehow misses the check on the active cells without the sleep
+    Sleep    5s
+    Wait Until JupyterLab Code Cell Is Not Active  timeout=120s
+    JupyterLab Code Cell Error Output Should Not Be Visible
+    SeleniumLibrary.Capture Page Screenshot
+
+Wait For Model Registry Containers To Be Ready
+    [Documentation]    Wait for model-registry-deployment to be ready
+    ${result}=    Run Process
+    ...        oc wait --for\=condition\=Available --timeout\=5m -n ${PRJ_TITLE} deployment/model-registry-db
+    ...        shell=true    stderr=STDOUT
+    Log To Console    ${result.stdout}
+    ${result}=    Run Process
+    ...        oc wait --for\=condition\=Available --timeout\=5m -n ${PRJ_TITLE} deployment/model-registry-deployment
+    ...        shell=true    stderr=STDOUT
+    Log To Console    ${result.stdout}
```

Comment:
maybe you could use `Wait For Pods Status` or `Wait For Deployment Replica To Be Ready` or `Wait For Pods To Be Ready`

Reply:
will be refactored later